examples/test_util/EGrid.cpp

```diff
@@ -0,0 +1,190 @@
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   OPM is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include "EGrid.hpp"
+
+#include <opm/common/ErrorMacros.hpp>
+#include <string>
+#include <string.h>
+#include <sstream>
+#include <iterator>
+#include <iomanip>
+#include <algorithm>
+#include <numeric>
+
+
+EGrid::EGrid(const std::string &filename) : EclFile(filename)
+{
+   loadData();
+
+   auto gridhead = get<int>("GRIDHEAD");
+
+   nI = gridhead[1];
+   nJ = gridhead[2];
+   nK = gridhead[3];
+
+   if (hasKey("ACTNUM")) {
+       auto actnum = get<int>("ACTNUM");
+
+       nactive = 0;
+       for (unsigned int i = 0; i < actnum.size(); i++) {
+           if (actnum[i] > 0) {
+               act_index.push_back(nactive);
+               glob_index.push_back(i);
+               nactive++;
+           } else {
+               act_index.push_back(-1);
+           }
+        }
+   } else {
+       int nCells = nI * nJ * nK;
+       act_index.resize(nCells);
+       glob_index.resize(nCells);
+       std::iota(act_index.begin(), act_index.end(), 0);
+       std::iota(glob_index.begin(), glob_index.end(), 0);
+   }
+
+   coord_array = get<float>("COORD");
+   zcorn_array = get<float>("ZCORN");
+}
+
+
+int EGrid::global_index(int i, int j, int k) const
+{
+    if ((i < 0) || (i >= nI) || (j < 0) || (j >= nJ) || (k < 0) || (k >= nK)) {
+        OPM_THROW(std::invalid_argument, "i, j or/and k out of range");
+    }
+
+    return i + j*nI + k*nI*nJ;
+}
+
+
+int EGrid::active_index(int i, int j, int k) const
+{
+    int n = i + j*nI + k*nI*nJ;
+
+    if ((i < 0) || (i >= nI) || (j < 0) || (j >= nJ) || (k < 0) || (k >= nK)) {
+        OPM_THROW(std::invalid_argument, "i, j or/and k out of range");
+    }
+
+    return act_index[n];
+}
+
+
+void EGrid::ijk_from_active_index(int actInd, int& i, int& j, int& k)
+{
+    if (actInd < 0 || actInd >= nactive) {
+        OPM_THROW(std::invalid_argument, "active index out of range");
+    }
+
+    int _glob = glob_index[actInd];
+
+    k = _glob / (nI*nJ);
+
+    int rest = _glob % (nI*nJ);
+
+    j = rest / nI;
+    i = rest % nI;
+}
+
+
+void EGrid::ijk_from_global_index(int globInd, int& i, int& j, int& k)
+{
+    if (globInd < 0 || globInd >= nI*nJ*nK) {
+        OPM_THROW(std::invalid_argument, "global index out of range");
+    }
+
+    k = globInd / (nI*nJ);
+
+    int rest = globInd % (nI*nJ);
+
+    j = rest / nI;
+    i = rest % nI;
+}
+
+
+void EGrid::getCellCorner(int i, int j, int k,
```

Comment:
The function fetches all eight corners - so I guess the plural form `getCellCorners` would be a better name?

Reply:
done.